src/datasources/erc721.ts

```diff
@@ -88,87 +94,89 @@ export function handleConsecutiveTransfer(event: ConsecutiveTransfer): void {
-		// ev.emitter     = contract.id
-		// ev.transaction = transactions.log(event).id
-		// ev.timestamp   = event.block.timestamp
-		// ev.token       = token.id
-		// ev.owner       = owner.id
-		// ev.approved    = approved.id
-		// ev.save()
-	}
+	if (contract == null) return
+
+	let token    = fetchERC721Token(contract, event.params.tokenId)
+	let owner    = fetchAccount(event.params.owner)
+	let approved = fetchAccount(event.params.approved)
+
+	token.owner    = owner.id // this should not be necessary, owner changed is signaled by a transfer event
+	token.approval = approved.id
+
+	token.save()
+	owner.save()
+	approved.save()
+
+	// let ev = new Approval(events.id(event))
+	// ev.emitter     = contract.id
+	// ev.transaction = transactions.log(event).id
+	// ev.timestamp   = event.block.timestamp
+	// ev.token       = token.id
+	// ev.owner       = owner.id
+	// ev.approved    = approved.id
+	// ev.save()
 }
 
 export function handleApprovalForAll(event: ApprovalForAllEvent): void {
 	let contract = fetchERC721(event.address)
-	if (contract != null) {
-		let owner      = fetchAccount(event.params.owner)
-		let operator   = fetchAccount(event.params.operator)
-		let delegation = fetchERC721Operator(contract, owner, operator)
-
-		delegation.approved = event.params.approved
-
-		delegation.save()
-
-		// 	let ev = new ApprovalForAll(events.id(event))
-		// 	ev.emitter     = contract.id
-		// 	ev.transaction = transactions.log(event).id
-		// 	ev.timestamp   = event.block.timestamp
-		// 	ev.delegation  = delegation.id
-		// 	ev.owner       = owner.id
-		// 	ev.operator    = operator.id
-		// 	ev.approved    = event.params.approved
-		// 	ev.save()
-	}
+	if (contract == null) return
+
+	let owner      = fetchAccount(event.params.owner)
+	let operator   = fetchAccount(event.params.operator)
+	let delegation = fetchERC721Operator(contract, owner, operator)
+
+	delegation.approved = event.params.approved
+
+	delegation.save()
+
+	// 	let ev = new ApprovalForAll(events.id(event))
+	// 	ev.emitter     = contract.id
+	// 	ev.transaction = transactions.log(event).id
+	// 	ev.timestamp   = event.block.timestamp
+	// 	ev.delegation  = delegation.id
+	// 	ev.owner       = owner.id
+	// 	ev.operator    = operator.id
+	// 	ev.approved    = event.params.approved
+	// 	ev.save()
 }
 
 export function handleMetadataUpdate(event: MetadataUpdateEvent) : void {
-	let erc721   = IERC721.bind(event.address)
 	let contract = fetchERC721(event.address)
+	if (contract == null) return
 
-	if (contract != null) {
-		if (contract.supportsMetadata) {
-			let token        = fetchERC721Token(contract, event.params._tokenId)
-			let try_tokenURI = erc721.try_tokenURI(event.params._tokenId)
-			token.uri        = try_tokenURI.reverted ? '' : try_tokenURI.value
-			token.save()
-		} else {
-			// add a warning ?
-		}
+	if (contract.supportsMetadata) {
+		_updateURI(contract, event.params._tokenId)
+	} else {
+		// add a warning ?
```

Comment:
I think this is a noop. Updating metadata to an unsupported token should just do nothing imo 🤔 

We're already ignoring the `else` side in [erc721.ts#fetchERC721Token](https://github.com/OpenZeppelin/openzeppelin-subgraphs/blob/bd3cb3d6b65dade98be85b3fee81657940f35c78/src/fetch/erc721.ts#L79)